src/utils.py

```diff
@@ -23,7 +23,8 @@ def ensure_parent_containers(tiled_uri, tiled_api_key):
     
     last_container = from_uri(tiled_root, api_key=tiled_api_key)
 
-    for part in path_parts:
+    container_parts = path_parts[3:]
```

Comment:
Suggestion: Please add a short comment somewhere indicating why [3:]

Reply:
The first 3 elements of `path_part` refers to the header `api/v1/metadata`.  Since this has been checked during authentication step above, it should be avoided in the following checks. Otherwise it will create duplicated containers of api, v1 and metadata